demisto_sdk/commands/common/singleton.py

```diff
@@ -1,7 +1,20 @@
-class Singleton:
+from abc import abstractmethod
+
+
+class PydanticSingleton:
     _instance = None
 
-    def __new__(cls, *args, **kwargs):
-        if not isinstance(cls._instance, cls):
-            cls._instance = super().__new__(cls)
+    @classmethod
+    def get_instance(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = cls.get_instance_from(*args, **kwargs)
         return cls._instance
+
+    @classmethod
+    @abstractmethod
+    def get_instance_from(cls, *args, **kwargs):
```

Comment:
in pydantic objects we most of the times want to initialize the class not from the constructor, but from a class method.

this is a general abstract method that each singelton pydantic object should inherit and implement it.  

i changed the way the singelton was implemented because it didn't work with pydnatic